libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayHistoryMapper.kt

```diff
@@ -130,14 +129,6 @@ class ReplayHistoryMapper private constructor(
          * @return [ReplayHistoryMapper]
          */
         fun build(): ReplayHistoryMapper {
-            if (setRouteMapper === DefaultSetRouteMapper) {
-                logW(
-                    "Mapper uses a deprecated default `setRouteMapper` based on " +
-                        "`ReplaySetRoute` type. Consider switching to a mapper based on " +
-                        "`ReplayNavigationSetRoute`.",
-                    "ReplayHistoryMapper"
-                )
-            }
```

Comment:
What was the reason for using a deprecated route mapper by default and why can we avoid it now?

Reply:
This warning and requirement were introduced with the NavigationRoute https://github.com/mapbox/mapbox-navigation-android/commit/9a785a4b8a0d4bf08d3a7b9869b0082c8823e9e0#diff-999c89fb6439fbdec91f89734b7f95947814c4fad7e9e0417727f34939a44688L185-L191

I can't find a reason to require a warning because the `directionsRoute` comes from the `navigationRoute` https://github.com/mapbox/mapbox-navigation-android/blob/09b52480c921d4acb03e206da778d4a4735a6477/libnavigation-core/src/main/java/com/mapbox/navigation/core/history/model/HistoryEventSetRoute.kt#L31-L37

Reply:
Could it be caused by the fact that the mapper should be able to parse old history files, which were recorded before `NavigationRoute` was introduced? I don't know if that makes sense, I'm just curious why we weren't able to do this change immediately.

Reply:
Actually the SEMVER reason is because it will emit `ReplaySetNavigationRoute` instead of `ReplaySetRoute`